python_modules/dagster/dagster/_core/definitions/declarative_automation/automation_condition.py

```diff
@@ -771,9 +772,32 @@ def __init__(
         self._end_timestamp = get_current_timestamp()
 
         # hidden_param which should only be set by legacy RuleConditions
-        self._subsets_with_metadata = check.opt_sequence_param(
+        subsets_with_metadata = check.opt_sequence_param(
             kwargs.get("subsets_with_metadata"), "subsets_with_metadata", AssetSubsetWithMetadata
         )
+        metadata = check.opt_mapping_param(
```

Comment:
I think I would have opted to go the other route, with the parallel metadata object.  The thinking mostly being that maybe we would eventually be able to kill the param and then the underlying data structure would match the legit params?

Reply:
I think you're totally right and I was making it more complicated for no reason -- updated